core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1784,8 +1784,7 @@ private long loadMetadataAndJournal(LockedInodePath inodePath, LoadMetadataOptio
     UnderFileSystem ufs = resolution.getUfs();
     try {
       if (!ufs.exists(ufsUri.toString())) {
-        throw new FileDoesNotExistException(
-            ExceptionMessage.PATH_DOES_NOT_EXIST.getMessage(path.getPath()));
+        return AsyncJournalWriter.INVALID_FLUSH_COUNTER;
```

Comment:
In that case is a better fix to not try to load metadata for directories which are not persisted or files which are already in Alluxio?